dbt_automation/utils/bigquery.py

```diff
@@ -241,4 +241,18 @@ def get_total_rows(self, schema: str, table: str) -> int:
             return total_rows
         except Exception as e:
             logger.error(f"Failed to fetch total rows for {schema}.{table}: {e}")
+            raise
+
+    def get_column_data_types(self) -> list:
+        """Returns a list of distinct column data types from BigQuery."""
+        try:
+            resultset = self.execute(
+                """
+                SELECT DISTINCT data_type
+                FROM `project_id.dataset_id.INFORMATION_SCHEMA.COLUMNS`
```

Comment:
this works? you don't need to provide the actual project ID and dataset ID?

Reply:
even with the project ID and dataset ID, the query only returns the types currently used in the tables of that schema... not all supported BQ types

Reply:
couldn't find a query that fetches all the columns that BQ supports

Reply:
Yes, this I saw. One way is to send this list of datatypes directly instead of a query. Should we do it like that?

Reply:
yes if a query doesn't exist then their official documentation is next best